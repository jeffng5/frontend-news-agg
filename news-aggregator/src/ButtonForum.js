import React, { useState} from 'react'
import './css/FrontPage.css'
import {Helpers} from "./helpers"



const ButtonForum = ({username, url, title, description, author, urlToImage}) => {
  
  
    const [state, setState] = useState([])
    const [buttonColor, setButtonColor] = useState(true)


    const handleForum = async() => {
        try {
            setButtonColor(false)
            //using function to make backend API call to POST saved article
            const res = await Helpers.postForum(username, url, title, description,author, urlToImage)
            setState(res)
        
        }
        catch (e){
            console.log(e)
        }
    }
    

    return (
        <button className='forum' onClick={handleForum} style={{backgroundColor: buttonColor ? 'gold' : 'grey'  }}><h6 id='button'>Post to Forum</h6></button>
    )
    }

    export default ButtonForum