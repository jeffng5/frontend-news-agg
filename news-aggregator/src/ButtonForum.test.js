import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ButtonForum from './ButtonForum'

it('renders without crashing', function() {
  render(<ButtonForum />);
})

//snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(<ButtonForum />);
    expect(asFragment()).toMatchSnapshot();
  });

test('testing component', ()=> {
    const {getByText} = render( <ButtonForum />);
    getByText('Post to Forum')
})
