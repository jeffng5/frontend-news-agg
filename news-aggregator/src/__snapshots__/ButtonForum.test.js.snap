// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="forum"
    style="background-color: aquamarine;"
  >
    <h6
      id="button"
    >
      Post to Forum
    </h6>
  </button>
</DocumentFragment>
`;
